types/scrivito/index.d.ts

```diff
@@ -101,71 +102,73 @@ interface BackgroundImageTagProps {
     style: CSSPropsWithoutBackground & BackgroundImageBackgroundProp;
     className?: string;
 }
-export class BackgroundImageTag extends Component<BackgroundImageTagProps, any> { }
+export class BackgroundImageTag extends React.Component<BackgroundImageTagProps, any> { }
 
 interface ChildListTagProps {
     parent?: Obj;
     tag?: string;
-    // TODO: hard to type
-    renderChild?: (child: any) => any;
+    renderChild?: (child: Obj) => React.ReactElement;
 }
 
-export class ChildListTag extends Component<ChildListTag, any> { }
+export class ChildListTag extends React.Component<ChildListTagProps, any> { }
 
 interface ContentTagProps extends React.HTMLAttributes<any> {
     attribute: string;
-    content: Obj | Widget;
+    content?: Obj | Widget;
     tag?: string;
+    page?: any;
     widgetProps?: object;
 }
 
-export class ContentTag extends Component<ContentTagProps, any> { }
+export class ContentTag extends React.Component<ContentTagProps, any> { }
 
-export class CurrentPage extends Component<{}, any> { }
+export class CurrentPage extends React.Component<{}, any> { }
 
 interface ImageTagProps extends React.HTMLAttributes<HTMLImageElement> {
     attribute: string;
     content: Binary | Obj | Widget;
     alt?: string;
 }
 
-export class ImageTag extends Component<ImageTagProps, any> { }
+export class ImageTag extends React.Component<ImageTagProps, any> { }
 
-export class InPlaceEditingOff extends Component<any, any> { }
+export class InPlaceEditingOff extends React.Component<any, any> { }
 
 interface LinkTagProps extends React.HTMLAttributes<HTMLAnchorElement> {
     to: Obj | Link;
     params?: object;
     onClick?: (event: React.MouseEvent) => void;
 }
-export class LinkTag extends Component<LinkTagProps, any> { }
+export class LinkTag extends React.Component<LinkTagProps, any> { }
 
-export class NotFoundErrorPage extends Component<{}, any> { }
+export class NotFoundErrorPage extends React.Component<{}, any> { }
 
-export class RestoreInPlaceEditing extends Component<{}, any> { }
+export class RestoreInPlaceEditing extends React.Component<{}, any> { }
 
 interface WidgetTagProps extends React.HTMLAttributes<any> {
-    tag: string;
+    tag?: string;
     [key: string]: any;
 }
 
-export class WidgetTag extends Component<WidgetTagProps, any> { }
+export class WidgetTag extends React.Component<WidgetTagProps, any> { }
 
 /**
  * Config definitions
  */
 
 interface ConfigOptions {
     tenant: string;
-    homepage: () => Obj;
+    homepage?: () => Obj | null;
```

Comment:
Just to confirm, you wanted `() => (Obj | null)` and not `(() => Obj) | null`, right?

Reply:
Yea, sadly the function is entirely optional but may return "null", too.